saleor/schedulers/schedulers.py

```diff
@@ -0,0 +1,170 @@
+import copy
+import logging
+import time
+from typing import List, NamedTuple
+
+import celery.beat
+import celery.schedules
+from celery.signals import setup_logging
+from django_celery_beat import models as base_models
+from django_celery_beat.clockedschedule import clocked
+from django_celery_beat.schedulers import DatabaseScheduler as BaseDatabaseScheduler
+from django_celery_beat.schedulers import ModelEntry
+
+from . import customschedule, models
+
+logger = logging.getLogger(__name__)
+
+
+@setup_logging.connect
+def setup_celery_logging(loglevel=None, **_kwargs):
+    """Configure the logging level of module loggers with the flag passed to Celery.
+
+    Respects the --loglevel=<level> passed to Celery beats
+    """
+    logging.getLogger("saleor.schedulers").setLevel(loglevel)
+
+
+def is_numeric_value(value):
+    return isinstance(value, (int, float)) and not isinstance(value, bool)
+
+
+class HeapEventType(NamedTuple):
+    time: float  # timestamp when it should trigger next
+    priority: int
+    entry: celery.beat.ScheduleEntry
+
+
+class CustomModelEntry(ModelEntry):
+    """Scheduler entry taken from database row."""
+
+    model_schedules = (
+        (celery.schedules.crontab, base_models.CrontabSchedule, "crontab"),
+        (celery.schedules.schedule, base_models.IntervalSchedule, "interval"),
+        (celery.schedules.solar, base_models.SolarSchedule, "solar"),
+        (clocked, base_models.ClockedSchedule, "clocked"),
+        (customschedule.CustomSchedule, models.CustomSchedule, "custom"),
+    )
+    save_fields = ["last_run_at", "total_run_count", "no_changes"]
+
+    @classmethod
+    def from_entry(cls, name, app=None, **entry):
+        # Super method has 'PeriodicTask' hardcoded
+        return cls(
+            models.CustomPeriodicTask._default_manager.update_or_create(
+                name=name,
+                defaults=cls._unpack_fields(**entry),
+            ),
+            app=app,
+        )
+
+
+class BaseScheduler(celery.beat.Scheduler):
+    """Define the base scheduler for Celery beat."""
+
+    def tick(
```

Comment:
Original code is here: https://github.com/celery/celery/blob/aa9fd8a6c06e69c7eda2a59866c3d84622c85d20/celery/beat.py#L321-L359

Note that `is_numeric_value` is new at [v5.3.0a1](https://github.com/celery/celery/releases/tag/v5.3.0a1) thus can't be imported yet